### ISSUE TITLE
[a11y]: OverFlowMenuItem content is not within a landmark element

### ISSUE THREAD

**Issue by markkulube** [carbon-design-system/carbon#10044]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.46.0

### Automated testing tool and ruleset

IBM Equal Accessibility Checker

### Assistive technology

No

### Description

The underlying DOM elements of the `OverFlowMenuItem` are not within a landmark element. In particular [the div at FloatingMenu.js#L470](https://github.com/carbon-design-system/carbon/blob/84de2dc35cb337b9e7fd60e7fe0535449ae730ce/packages/react/src/internal/FloatingMenu.js#L470) could be assigned `role="contentinfo"` to suppress the violation.
<img width="731" alt="Screen Shot 2021-11-03 at 12 36 21 PM" src="https://user-images.githubusercontent.com/34955942/140103559-cd94340a-1043-4a5a-a04a-7825c5d04e09.png">

The violation would be addressed with any landmark role  but the challenge is that this is a portal use-case and so far only `role="dialog"` is exempt from being in a landmark role.

The FloatingMenu is rendered using a React Portal which triggers the landmark violation. This is rendered as a portal to address the clipping issues that teams run into. This means that it is removed from the content flow and placed at the end of the document outside of the tree. The component itself uses role="menu" with focus sentinels to trap focus  - these are probably what prompt the violation

`aria-owns` should address the issue because the owning component should be within the landmark. So, per the accessibility tree, the `FloatingMenu` will also be within that landmark (even though it doesn’t appear so in the DOM). There might have to be an update to the rule to cover that situation though.

So basically…

```
landmark
  - button
  - div owns menu
floating menu
```

resolves in the accessibility tree to

```
landmark
  - button
  - floating menu
```

Landmark roles provide programmatic access to sections of a web page, making it easy for the user to understand the structure of the page and easily navigate to different sections. They also enable users to skip over blocks of content that are repeated on multiple pages.

![Screen Shot 2021-11-03 at 11 58 58 AM](https://user-images.githubusercontent.com/34955942/140096733-3a433c27-8ae6-41cd-a052-c026542980ea.png)

### CodeSandbox example

https://gfegr.csb.app/

### Steps to reproduce

1. Click OverFlowMenu
2. Right click any OverFlowMenuItem
3. Open dev tools (inspect)
4. Open accessibility checker
5. Click scan

![Screen Shot 2021-11-03 at 11 58 58 AM](https://user-images.githubusercontent.com/34955942/140097039-5c7b4cc7-41fc-4607-bed3-47d866dda468.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Hey there @markkulube, I'm unable to reproduce the issue in our [storybook](https://react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&args=&viewMode=story)

<img width="1109" alt="Screen Shot 2022-02-28 at 12 49 52 PM" src="https://user-images.githubusercontent.com/11928039/156032905-98f8c93d-77c7-479a-aa39-33363556f457.png">

We use `data-floating-menu-container` to render the overflow menu relevant to the container it is placed in. If one is not provided, then defaults to being placed last in the DOM (and throwing these violations). It also needs to be placed inside a properly landmarked region. Here is an example:

https://codesandbox.io/s/k2yoon?file=/src/index.js

Let me know if that works 👍🏻

**Comment by markkulube** [comment]
@tw15egan thank you. 

We resolved with the technique: by adding`data-floating-menu-container="true"` to the top most container.

closed in PR: https://github.ibm.com/Bluemix/Bluemix.Dashboard/pull/2266

**Comment by markkulube** [comment]
@tw15egan thank you. 

We resolved with the technique: by adding`data-floating-menu-container="true"` to the top most container.

closed in  [resourse-list/pr#2266](https://github.ibm.com/Bluemix/Bluemix.Dashboard/pull/2266)
